src/EVM/SymExec.hs

```diff
@@ -192,7 +192,7 @@ abstractVM
   -> ST s (VM s)
 abstractVM cd contractCode maybepre create = do
   let value = TxValue
-  let code = if create then InitCode contractCode mempty else RuntimeCode (ConcreteRuntimeCode contractCode)
+  let code = if create then InitCode contractCode (fst cd) else RuntimeCode (ConcreteRuntimeCode contractCode)
```

Comment:
hm, I'm kind of in two minds here, this is obviously a very clean change, but it's also a bit of an abuse of semantics since calldata isn't really a thing that makes sense for constructors. 

If we go this route we should also maybe make sure to avoid writing calldata to the vm state if `create` is true?